### PR TITLE
[Schedulers] Add beta sigmas / beta noise schedule

### PR DESCRIPTION
**Pull request by hlky**
# What does this PR do?

This PR adds [beta sigmas](https://github.com/lllyasviel/stable-diffusion-webui-forge/blob/a82d5d177c065f8a6251850e233dca97f15c13ac/modules/sd_schedulers.py#L119-L127) from [Beta Sampling is All You Need](https://huggingface.co/papers/2407.12173). Added only to Euler for now, after initial review we will add it to other schedulers. Documentation will be updated for this scheduler and any others implemented in a subsequent PR after the initial review and after they've been added to other schedulers.

The implementation is quite simple, we add `use_beta_sigmas` parameter and `_convert_to_beta` function.

Beta sigmas depends on `scipy` so we add a check with `is_scipy_available()`.

The check for `self.config.use_exponential_sigmas and self.config.use_karras_sigmas` is moved to schedule instantiation and changed to use `sum([self.config.use_beta_sigmas, self.config.use_exponential_sigmas, self.config.use_karras_sigmas]) > 1`, any future noise schedules can be added to the list to ensure only one can be selected.

As a note for reviewers: `alpha` and `beta` in `_convert_to_beta` are set to the default values [used by Forge](https://github.com/lllyasviel/stable-diffusion-webui-forge/blob/a82d5d177c065f8a6251850e233dca97f15c13ac/modules/shared_options.py#L413-L414), should we make these configurable so we can set e.g. `beta_sigmas_alpha=1.0`, `beta_sigmas_beta=1.0`? I don't know how common it is to change these values. This is also relevant to future `polyexponential` sigmas, `polyexponential` produces the same as `exponential` unless the `rho` value is changed, again I don't know how common it is to change this value, if we're not going to make the values configurable we don't need to add `polyexponential`. However it would be nice to have 1:1 compatibility with the webuis.

Example usage:
```python
from diffusers import StableDiffusionXLPipeline, EulerDiscreteScheduler
import torch

pipeline: StableDiffusionXLPipeline = StableDiffusionXLPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0",
    variant="fp16",
    torch_dtype=torch.float16,
)
pipeline.scheduler = EulerDiscreteScheduler.from_config(
    pipeline.scheduler.config, timestep_spacing="linspace", use_beta_sigmas=True
)
```

Note that we set `timestep_spacing="linspace"` in addition to `use_beta_sigmas=True`, this is because XL's scheduler config uses `timestep_spacing="leading"` by default and the intent is to match results from the webuis where these noise schedules are currently used. We may wish to override `timestep_spacing` to `linspace` when `use_beta_sigmas=True` or add a warning.

Tested against [Forge](https://github.com/lllyasviel/stable-diffusion-webui-forge) (note that `cuda` device for generator is essential to match Forge results):
```python
generator = torch.Generator("cuda").manual_seed(49136503742430)

image = pipeline(
    prompt="enormous kirby. space background",
    num_inference_steps=20,
    guidance_scale=5.0,
    generator=generator,
).images[0]
image
```
output

and with the same settings on Forge we get the same result:

The output is similar yet better than exponential #9499 note the shape of Kirby's feet and the planets.

Both beta and exponential are better than the default noise schedule with XL:
output2

#9490 

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
cc @asomoza @yiyixuxu

### CODE REVIEW

File: src/diffusers/schedulers/scheduling_euler_discrete.py [hunk 1]
```diff
@@ -17,10 +17,11 @@
 from typing import List, Optional, Tuple, Union
 
 import numpy as np
+import scipy.stats
```
Comment (yiyixuxu) [inline src/diffusers/schedulers/scheduling_euler_discrete.py:20]: so this I think would cause an error if user does not have scipy installed
1. can we look into if we can use beta sigmas without scipy?
2. if not, let's add 
```
if is_scipy_available:
    import scipy.stats
```

File: src/diffusers/schedulers/scheduling_euler_discrete.py [hunk 2]
```diff
@@ -502,6 +518,37 @@ def _convert_to_exponential(self, in_sigmas: torch.Tensor, num_inference_steps:
...
+
+        if hasattr(self.config, "sigma_max"):
+            sigma_max = self.config.sigma_max
+        else:
+            sigma_max = None
+
+        sigma_min = sigma_min if sigma_min is not None else in_sigmas[-1].item()
+        sigma_max = sigma_max if sigma_max is not None else in_sigmas[0].item()
+
+        sigmas = torch.FloatTensor(
```
Comment (yiyixuxu) [inline src/diffusers/schedulers/scheduling_euler_discrete.py:541]:
```suggestion
        sigmas = torch.Tensor(
```

**Review by yiyixuxu — COMMENTED** [review]
left a feedback! thanks!

File: src/diffusers/schedulers/scheduling_euler_discrete.py (hunk 1 above)
Comment (hlky) [reply src/diffusers/schedulers/scheduling_euler_discrete.py]: I'd have to look into how feasible it is to reimplement `scipy.stats.beta.ppf`.

I've added the check though, thanks!

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Comment by yiyixuxu** [comment]
we need `make style` now

**Comment by hlky** [comment]
I've applied `make style`.
